src/app/interaction-model/Events.h

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app/EventLoggingDelegate.h>
+#include <app/EventLoggingTypes.h>
+#include <app/MessageDef/EventDataIB.h>
+#include <app/data-model/Encode.h>
+#include <app/data-model/FabricScoped.h>
+#include <lib/core/CHIPError.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#include <type_traits>
+
+namespace chip {
+namespace app {
+namespace InteractionModel {
+
+namespace internal {
+template <typename T>
+class SimpleEventLoggingDelegate : public EventLoggingDelegate
```

Comment:
I duplicated that one becasue EventLogging.h is already in-use and tied to `src/app:app` gn target. I tried to ensure that existing interaction-model classes are reasonably stand-alone because coupling in `src/app:app` and `src/app:interaction-model`  are very coupled and uncoupling them seemed hard. Ialso originally wanted a simple interface however it does seem I ended up with it being identical.

I can merge them together, however I was thinking I may want a merge at a later time once we start using the new interaction-model bits.

Reply:
Merging later is fine, I guess, as long as it happens.